store/mongo/datastore_mongo_test.go

```diff
@@ -476,7 +476,7 @@ func TestMigrate(t *testing.T) {
 
 			// set up automigration
 			if tc.automigrate {
-				db = db.WithAutomigrate()
+				db = db.WithAutomigrate().(*DataStoreMongo)
```

Comment:
Can we somehow skip type assertion here?

Reply:
no, WithAutomigrate must be public, so part of the interface, so it must return the same interface.

aside from introducing another wrapper maybe, but I didn;t want this to happen.